text/0000-pack-cache-options.md

```diff
@@ -0,0 +1,224 @@
+
+The platform spec currently provides two formats for caching build layers. These formats are a directory or an image.
+
+Additionally, there is an option to specify a _`launch-cache`_ directory to cache launch layers.
+
+> NOTE: Currently _`launch-cache` flag_ is only used by the current implementation of the lifecycle when _`-daemon` flag_ is provided.
+
+Pack, as a consumer of the lifecycle using the platform spec, uses these options in the following ways:
+
+* By default, `pack build` would create two named volumes
+
+<!-- - Why should we do this?
+- What use cases does it support?
+- What is the expected outcome? -->
+
+##### Use Cases
+
+* As a Pack user, I would like to build using a docker daemon while storing _`build` cache_ as an image (aka _`cache-image`_) in a registry. ([buildpacks/pack#1024](https://github.com/buildpacks/pack/issues/1024#issuecomment-762901676))
+* As a Pack user using `pack` on BitBucket Pipelines, I would like to be able to store the _`build` cache_ and _`launch` cache_ on the file system. ([bitbucket discussion](https://community.atlassian.com/t5/Bitbucket-discussions/Using-CNCF-Buildpacks-on-Bitbucket-Pipelines/td-p/1516690))
+* As a Pack user, I would like to be able to specify the name of the volumes used by pack for caching. ([buildpacks/pack#665](https://github.com/buildpacks/pack/issues/665))
+
+# What it is
+[what-it-is]: #what-it-is
+
+<!-- This provides a high level overview of the feature.
+
+- Define any new terminology.
+- Define the target persona: buildpack author, buildpack user, platform operator, platform implementor, and/or project contributor.
+- Explaining the feature largely in terms of examples.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to existing users and new users. -->
+
+A new variadic flag that enables configuring different cache types geared towards app developers or operators that are using pack in very specific (and sometimes restrictive) ways that require them to configure caching in a more advanced fashion.
+
+The options available via this new flag would have sensible defaults based on the current behaiviour with the exception of the `--cache-image` which would be deprecated and later removed to reduce confusion. Overall UX would remain the same while allowing for greater flexibility.
+
+An additional benefit of this format is that it can later be expanded on if more options for a specific format were deamed necessary.
+
+```
+pack build --cache 'type=<type>;format=<format>;<additional-options>' --cache ...
+```
+
+### `cache` flag options
+* _**type**_ - The type of cache to configure [`build`, `launch`] (default: `build`).
+* _**format**_ - The format in which to store that cache [`bind`, `image`, `volume`] (default: `volume`). 
+
+###### Format specific options
+
+* `bind`
+    * _**source**_ - Path to the local directory to use as cache.
+* `image`
```

Comment:
And, to clarify (per https://github.com/buildpacks/pack/issues/1024#issuecomment-762901676), this image could be a local image, or would it still need to be a registry image?

Reply:
The linked comment does bring up the _future_ potential for an additional flag similar to [buildkit](https://github.com/moby/buildkit#output):

```
push=<boolean>
```

ie:

```
type=build;format=image;name=io.docker.io/myorg/my-cache:build;push=false
```